tools/execsnoop.py

```diff
@@ -193,19 +197,19 @@ def print_event(cpu, data, size):
     elif event.type == EventType.EVENT_RET:
         if args.fails and event.retval == 0:
             skip = True
-        if args.name and not re.search(args.name, event.comm):
+        if args.name and not re.search(bytes(args.name), event.comm):
             skip = True
-        if args.line and not re.search(args.line,
-                                       b' '.join(argv[event.pid]).decode()):
+        if args.line and not re.search(bytes(args.line),
+                                       b' '.join(argv[event.pid])):
             skip = True
 
         if not skip:
             if args.timestamp:
                 print("%-8.3f" % (time.time() - start_ts), end="")
             ppid = get_ppid(event.pid)
-            print("%-16s %-6s %-6s %3s %s" % (event.comm.decode(), event.pid,
-                    ppid if ppid > 0 else "?", event.retval,
-                    b' '.join(argv[event.pid]).decode()))
+            ppid = b"%d" % ppid if ppid > 0 else b"?"
+            printb(b"%-16s %-6d %-6s %3d %s" % (event.comm, event.pid,
+                   ppid, event.retval, b' '.join(argv[event.pid])))
```

Comment:
The simplest is just to leave it as `print("%s", event.comm)` which will give you `b'/usr/bin/bash'` or similar output (if run from python3).

For the other suggestions, the issue remains that in order to convert the byte string to a unicode string, some encoding must be chosen. Simply wrapping these in utf-8 can still fail (e.g. try `b"\xff".decode("utf8")`), so I don't see that path working out.